Makefile

```diff
@@ -968,14 +986,14 @@ $(TARGET_ELF):  $(TARGET_OBJS)
 # Compile
 $(OBJECT_DIR)/$(TARGET)/%.o: %.c
 	$(V1) mkdir -p $(dir $@)
-	$(V1) $(if $(findstring $(subst ./src/main/,,$<), $(SPEED_OPTIMISED_SRC)), \
+	$(V1) $(if $(findstring $(notdir $<),$(SPEED_OPTIMISED_SRC)), \
```

Comment:
Aha, with the "subst ./src/main/,," construct, we will only have optimization of file under "src/main", not any of the "lib/main" files etc. I.e. not everything as before for F4. Need to think of something else here, or I will keep the PR as is. When the name clashes becomes a real problem, we need to fix that some other way. It is not really a problem now.